pydis_site/apps/resources/views/resources.py

```diff
@@ -1,7 +1,128 @@
-from django.views.generic import TemplateView
+import json
+import typing as t
+from pathlib import Path
 
+import yaml
+from django.core.handlers.wsgi import WSGIRequest
+from django.http import HttpResponse, HttpResponseNotFound
+from django.shortcuts import render
+from django.views import View
 
-class ResourcesView(TemplateView):
-    """View for resources index page."""
+from pydis_site import settings
+from pydis_site.apps.resources.templatetags.as_css_class import as_css_class
 
-    template_name = "resources/resources.html"
+RESOURCES_PATH = Path(settings.BASE_DIR, "pydis_site", "apps", "resources", "resources")
+
+
+class ResourceView(View):
+    """Our curated list of good learning resources."""
+
+    @staticmethod
+    def _sort_key_disregard_the(tuple_: tuple) -> str:
+        """Sort a tuple by its key alphabetically, disregarding 'the' as a prefix."""
+        name, resource = tuple_
+        name = name.casefold()
+        if name.startswith("the ") or name.startswith("the_"):
+            return name[4:]
+        return name
+
+    def __init__(self, *args, **kwargs):
+        """Set up all the resources."""
+        super().__init__(*args, **kwargs)
+
+        # Load the resources from the yaml files in /resources/
+        self.resources = {
+            path.stem: yaml.safe_load(path.read_text())
+            for path in RESOURCES_PATH.rglob("*.yaml")
+        }
+
+        # Sort the resources alphabetically
+        self.resources = dict(sorted(self.resources.items(), key=self._sort_key_disregard_the))
+
+        # Parse out all current tags
+        resource_tags = {
+            "topics": set(),
+            "payment_tiers": set(),
+            "difficulty": set(),
+            "type": set(),
+        }
+        for resource_name, resource in self.resources.items():
+            css_classes = []
+            for tag_type in resource_tags.keys():
+                # Store the tags into `resource_tags`
+                tags = resource.get("tags", {}).get(tag_type, [])
+                for tag in tags:
+                    tag = tag.title()
+                    tag = tag.replace("And", "and")
+                    resource_tags[tag_type].add(tag)
+
+                # Make a CSS class friendly representation too, while we're already iterating.
+                for tag in tags:
+                    css_tag = f"{tag_type}-{tag}"
+                    css_tag = css_tag.replace("_", "-")
+                    css_tag = css_tag.replace(" ", "-")
+                    css_classes.append(css_tag)
```

Comment:
Why aren't we just using the as_css_class template tag here? This is just duplicating code from there.